mxcube3/__init__.py

```diff
@@ -98,18 +98,22 @@ def exception_handler(e):
 
 
 def kill_processes():
-    pid_list = []
+    # Killing the processes causes pytest to fail because
+    # of non zero exit code, so we dont kill the processes
+    # when running the tests
+    if not server.testing:
+        pid_list = []
```

Comment:
No need to declare pid_list here, the code below takes care of it already.

Reply:
Yes, indeed but I actually prefer to declare variables like this, I know that it's maybe not very pytonic but it least indicates what the intended type is. It's true that one waste one assignment. I guess the name already indicates that it should be a list so I will remove it this time :)